### DISCUSSION TITLE
next is not defined

### DISCUSSION THREAD

**Discussion by bradrisse** [vercel/next.js#16225] [Help]
I have an intermittent issue on Google App Engine on a standard environment. The displayed browser error is 

```
An unexpected error has occurred.
```

The console error is:

```
main-e9fcf9c93d6d8ac0e441.js:formatted:406 ReferenceError: next is not defined
    at Object.Pmoa (index.js:formatted:3121)
    at l (webpack-1c5199ff66550d26e499.js:formatted:40)
    at Module.RNiq (index.js:formatted:3599)
    at l (webpack-1c5199ff66550d26e499.js:formatted:40)
    at index.js:formatted:6381
    at main-e9fcf9c93d6d8ac0e441.js:formatted:1018
    at e.value (main-e9fcf9c93d6d8ac0e441.js:formatted:1033)
    at U (main-e9fcf9c93d6d8ac0e441.js:formatted:208)
    at Array.map (<anonymous>)
    at Object.IKlv (main-e9fcf9c93d6d8ac0e441.js:formatted:210)
te @ main-e9fcf9c93d6d8ac0e441.js:formatted:406
```

The production build line main-e9fcf9c93d6d8ac0e441.js:formatted:406
```
    Pmoa: function(e, t) {
        e.exports = next
    },
```

Building nextjs in development mode leads to this line

```
/***/ "Pmoa":
/***/ (function(module, exports) {

module.exports = require("next");
```

Upon refresh the issue is resolved. 

My thoughts are that nextjs is being loaded after this line due to a cache issue or something else. I've tried to run down the cause to no avail. Any help or direction would be greatly appreciated.

**Comment by jamesmosier** [comment]
I'm not sure anything would be cache related, but how are you installing Next on your Google App Engine instance? Just via `npm install` (or `yarn install`) in your directory with your `package.json`? Have you checked to ensure Next is **not** installed globally too?

**Reply by bradrisse** [reply]
This is how the application is installed on google app engine https://github.com/bradrisse/nextjs-9-gae-demo. I'm also using 9.4.1, will try the upgrade. 

It's very difficult to replicate, essentially through using the application over time I will encounter it.

**Comment by meakyl** [comment]
We have the same issue with a classic Next app on Chrome and on production mode only (version 9.4.1)
After next build and node server.js with a custom express server.
After an CTRL+F5, if we navigate through the application we have this error then hit F5 and everything is ok. If I do a CTRL+F5 again, next navigation breaks with this error : "ReferenceError: next is not defined".
I can't make it break in dev mode though.

**Reply by jamesmosier** [reply]
It might be worth upgrading to the latest version of Next (`9.5.2`) just to be certain this issue wasn't fixed in a release since `9.4.1`.

Aside from that, do you happen to have a repository you could share that reproduces this? It doesn't have to be the exact repo, but just one that recreate this issue.
